test/scripts/traffic-flow-tests.sh

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Set default values
+export KUBECONFIG="${KUBECONFIG:-${HOME}/ovn.conf}"
+export OCI_BIN="${KIND_EXPERIMENTAL_PROVIDER:-docker}"
+export TFT_TEST_IMAGE="ghcr.io/wizhaoredhat/ocp-traffic-flow-tests:latest"
+export TRAFFIC_FLOW_TESTS_DIRNAME="ocp-traffic-flow-tests"
+export TRAFFIC_FLOW_TESTS_REPO="https://github.com/wizhaoredhat/ocp-traffic-flow-tests.git"
+export TRAFFIC_FLOW_TESTS_COMMIT="91bc034f71bf8e5e2009413754f20ac60442df65"
+export TRAFFIC_FLOW_TESTS="${TRAFFIC_FLOW_TESTS:-'1-3'}"
+
+export SCRIPT_DIR="$(cd -- "$(dirname -- "${BASH_SOURCE[0]}")" &> /dev/null && pwd)"
+cd "$(git rev-parse --show-toplevel)/.."
+export TOP_DIR="$(pwd)"
+export TRAFFIC_FLOW_TESTS_FULL_PATH="${TOP_DIR}/${TRAFFIC_FLOW_TESTS_DIRNAME}"
+
+log() {
+    local level="$1"
+    shift
+    echo "[$(date '+%Y-%m-%d %H:%M:%S')] [$level] $*"
+}
+
+error_exit() {
+    log "ERROR" "$*"
+    exit 1
+}
+
+check_dependencies() {
+    local dependencies=(sed awk pip git kubectl kind jq "${OCI_BIN}")
+    for cmd in "${dependencies[@]}"; do
+        if ! command -v "${cmd}" &> /dev/null; then
+            error_exit "Dependency not met: ${cmd}"
+        fi
+    done
+}
+
+usage() {
+    echo "Usage: $(basename "$0") {setup|run}"
+}
+
+create_config_file_defaults() {
+    local file_name="${1:-config.yaml}"
+
+    cat <<EOT > "$file_name"
+tft:
+  - name: "Github Workflow Test"
+    namespace: "default"
+    test_cases: "TBD"
+    duration: "10"
+    connections:
+      - name: "Connection_1"
+        type: "iperf-tcp"
+        instances: 1
+        server:
+          - name: "ovn-worker"
+            persistent: "false"
+            sriov: "false"
+        client:
+          - name: "ovn-worker2"
+            sriov: "false"
+        plugins: []
+kubeconfig:
+kubeconfig_infra:
+EOT
+}
+
+process_test_results() {
+    local result_file="$1"
+    local unique_tests_script='.["tft-tests"][]["flow_test"].tft_metadata.test_case_id'
+    local failures_script='
+      .["tft-tests"][] |
+      if .["flow_test"].success then empty
+      else {test_id: .["flow_test"].tft_metadata.test_case_id, message: .["flow_test"].msg}
+      end
+    '
+
+    log "INFO" "Processing test results from: ${result_file}"
+
+    local unique_test_ids
+    unique_test_ids=$(jq -r "${unique_tests_script}" "${result_file}" | sort -u)
+
+    echo "Test Cases:"
+    echo "-----------------------------------------------------"
+    echo "${unique_test_ids}"
+
+    local failures
+    failures=$(jq -c "${failures_script}" "${result_file}")
+
+    if [ -z "${failures}" ]; then
```

Comment:
`ocp-traffic-flow-tests` should make this easier.

There is already [print_results.py](https://github.com/wizhaoredhat/ocp-traffic-flow-tests/blob/91bc034f71bf8e5e2009413754f20ac60442df65/print_results.py). Note that this file operates on the output of `evaluator.py`, which compares the result to the thresholds.

The point is, I think we should add a script that parses the result in the ways we would want. If the current `print_result.py` doesn't do that, we should adjust it (or add another script).

Reply:
Oh, cool. Can you please teach me how to use it or point me to some documentation on how to use it?
I tried and failed miserably, before just making a simple json parsing script.

Reply:
Optimally, the `--help` options explains you all you want to know. If something is unclear, we should improve those.

The gist is:

`./main.py` [`CONFIG`](https://github.com/wizhaoredhat/ocp-traffic-flow-tests/blob/91bc034f71bf8e5e2009413754f20ac60442df65/config.yaml) generates those `./ft-logs/$DATE.json` output files.

Optionally, you can call `./evaluator.py` [`EVAL_CONFIG`](https://github.com/wizhaoredhat/ocp-traffic-flow-tests/blob/91bc034f71bf8e5e2009413754f20ac60442df65/eval-config.yaml) `./ft-logs/$DATE.json` `./ft-logs/eval-output`, which takes the output from the previous step and the `EVAL_CONFIG` with thresholds, and generates another output format. Theoretically, the thresholds in `EVAL_CONFIG` depends on each test environment, that's why they are separate.

Btw, if you called `./main.py CONFIG EVAL_CONFIG`, then the `evaluator.py` step would have been done in one step. In that case, the corresponding result files would be called `./ft-logs/$DATE.RESULT`.

Finally, there is `print_results.py` that pretty prints the results file from `evaluator.py`. It also exists with non-zero on any failures.

Currently `print_results.py` can only operate on the output files from `evaluator.py` -- which you don't have in your script so far. Maybe those two formats should be unified, or maybe `print_results.py` should work with both.

---

The point is, this bash script should become simpler. `process_test_results()` seems to do something that should be provided by ocp-tft itself. Maybe by adding an option to `print_results.py` to control what is printed there. Patch or suggestion welcome.

Reply:
@thom311 thank you for the review! I made some changes to account for the feedback. PTAL.

Reply:
Talked with @thom311 and @SalDaniele offline. Marking this as resolved.